src/test/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantOverlapAnnotatorUnitTest.java

```diff
@@ -91,6 +91,29 @@ public void testCreateWithSpecialNames() {
         tests.add(new Object[]{callNoIDAC, Arrays.asList(dbSNP_AC_AG), "rsID1;rsID3", true});
         tests.add(new Object[]{callNoIDAT, Arrays.asList(dbSNP_AC_AG), VCFConstants.EMPTY_ID_FIELD, false});
 
+        //multiple rsid which match multiallelic site
+        final VariantContext callNoIDT_C_TAC = makeVC("call", VCFConstants.EMPTY_ID_FIELD, Arrays.asList("T", "C", "TAC"));
+
+        final VariantContext dbSNP_T_TAC_TATAC = makeVC("DBSNP", "rsID1", Arrays.asList("T", "TAC", "TATAC"));
+        final VariantContext dbSNP_T_C = makeVC("DBSNP", "rsID2", Arrays.asList("T", "C"));
+
+        tests.add(new Object[]{callNoIDT_C_TAC, Arrays.asList(dbSNP_T_TAC_TATAC, dbSNP_T_C), "rsID1;rsID2", true});
+        tests.add(new Object[]{callNoIDT_C_TAC, Arrays.asList(dbSNP_T_C,dbSNP_T_TAC_TATAC), "rsID2;rsID1", true});
+
+        //mixed multiallelic in dbsnp
+        final VariantContext callNOID_T_TTCC = makeVC("call", VCFConstants.EMPTY_ID_FIELD, Arrays.asList("T", "TTCC"));
+
+        final VariantContext dbSNP_complex_mixed_site = makeVC("DBSNP", "rsID1", Arrays.asList("TTCCTCCTCCTCCTCCTCC", "T", "TTCCTCCTCCTCCTCCTCCTCC"));
+
+        tests.add(new Object[]{callNOID_T_TTCC, Arrays.asList(dbSNP_complex_mixed_site), "rsID1", true});
+
```

Comment:
Can you add a test with a DBSNP annotation with a particular length deletion and a variant that is also a deletion but with a different length to the annotation deletion? I have a hunch it will end up adding the annotation erroneously.

Reply:
Also test for insertions of various lengths.

Reply:
added tests for both, they seem to be behaving correctly.